streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -790,10 +814,17 @@ private UserTaskAssignmentListener assignTasksToClients(final Cluster fullMetada
             final org.apache.kafka.streams.processor.assignment.TaskAssignor assignor = userTaskAssignor.get();
             final TaskAssignment taskAssignment = assignor.assign(applicationState);
             final AssignmentError assignmentError = validateTaskAssignment(applicationState, taskAssignment);
-            processStreamsPartitionAssignment(clientMetadataMap, taskAssignment);
-            userTaskAssignmentListener = (assignment, subscription) -> assignor.onAssignmentComputed(assignment, subscription, assignmentError);
+            processStreamsPartitionAssignment(assignor, taskAssignment, assignmentError, clientMetadataMap, groupSubscription);
+            customTaskAssignmentListener = (assignment, subscription) -> {
+                assignor.onAssignmentComputed(assignment, subscription, assignmentError);
+                if (assignmentError != AssignmentError.NONE) {
+                    log.error("Task assignment returning empty GroupAssignment and failing due to error {}", assignmentError);
```

Comment:
The "returning empty GroupAssignment" part of the log is just talking about what we're passing in to the #onAssignmentComputed callback, which in this case is not an empty GroupAssignment.

```suggestion
                    log.error("Rebalance failed due to task assignor returning assignment with error {}", assignmentError);
```